### PR TITLE
Fix smart contract tests

### PR DESCRIPTION
**Pull request by andrewgordstewart**
We didn't run truffle tests for a long time, completely ignoring the
smart contracts. The application architecture changed significantly, so
the tests are updated accordingly.

The npm test scripts are re-jiggered. Truffle tests are now run by circleCI. To run application tests in watch mode, run `yarn test:app` instead.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,13 +56,15 @@
...
+    "test:ci": "run-s test:truffle 'test:app --all --ci'",
+    "test:app": "node scripts/test.js --env=jsdom",
+    "test:truffle": "run-s truffle:compile && (cd build/dist && pwd && ../../scripts/run-with-ganache.sh)",
     "postinstall": "rm -f node_modules/web3/index.d.ts",
     "copy_truffle_files": "copyfiles './contracts/**/*' './migrations/**/*' ./build/dist",
     "copy_dotenv": "copyfiles '.env' ./build/dist",
     "truffle:compile": "tsc && tsc -p ./tsconfig.truffle.json && run-s copy_truffle_files && (cd build/dist && ../../node_modules/.bin/truffle compile)",
-    "truffle:migrate": "run-s truffle:compile copy_dotenv && (cd build/dist && ../../node_modules/.bin/truffle migrate --network $TRUFFLE_NETWORK)",
-    "truffle:test": "run-s truffle:compile && (cd build/dist && pwd && ../../scripts/run-with-ganache.sh)"
+    "truffle:migrate": "run-s truffle:compile copy_dotenv && (cd build/dist && ../../node_modules/.bin/truffle migrate --network $TRUFFLE_NETWORK)"
```
Comment (lalexgap) [inline package.json]: It might be nice to default this to the development network if `$TRUFFLE_NETWORK` is not set?

**Comment by lalexgap** [comment]
We should probably update our tests to pull the contracts directly from the packages instead of depend on local files.

File: package.json (hunk 1 above)
Comment (andrewgordstewart) [reply package.json]: Oh yeah, nice catch. I only didn't do this because I was lazy at the time :P

**Comment by andrewgordstewart** [comment]
@lalexgap, `RockPaperScissorsGame` imports from the `fmg-core` package on npm, and not local files.

You might be confused, because there's a `fmg-rock-paper-scissors` package over there as well. But that package is on its way out, and should be removed.

**Pull request merged**